tst/EnergyPlus/unit/ScheduleManager.unit.cc

```diff
@@ -804,3 +806,445 @@ TEST_F(EnergyPlusFixture, Schedule_GetCurrentScheduleValue_DST)
+                        EXPECT_EQ(HourOfYear, ScheduleManager::GetCurrentScheduleValue(1));
+                    }
+                }
+            }
+        }
+
+        EXPECT_EQ(8784.0, HourOfYear);
+    }
+
+    {
+        DataEnvironment::DSTIndicator = 1; // DST IS ON
+        DataEnvironment::DayOfWeek = 0;
+        DataEnvironment::DayOfWeekTomorrow = 1;
+
+        Real64 HourOfYear = 0.0;
+        for (int month=1; month <= 12; ++month) {
+            DataEnvironment::Month = month;
+            for (int day = 1; day <= EndDayOfMonth(month); ++day) {
+                DataEnvironment::DayOfMonth = day;
+                ++DataEnvironment::DayOfWeek;
+                if (DataEnvironment::DayOfWeek > 7) {
+                    DataEnvironment::DayOfWeek = 1;
+                }
+                ++DataEnvironment::DayOfWeekTomorrow;
+                if (DataEnvironment::DayOfWeekTomorrow > 7) {
+                    DataEnvironment::DayOfWeekTomorrow = 1;
+                }
+
+                DataEnvironment::DayOfYear_Schedule = General::OrdinalDay(DataEnvironment::Month, DataEnvironment::DayOfMonth, 1);
+
+                for (int hr = 1; hr <= 24; ++hr) {
+                    ++HourOfYear;
+                    DataGlobals::HourOfDay = hr;
+                    for (int ts = 1; ts <= 4; ++ts) {
+                        DataGlobals::TimeStep = ts;
+
+                        ScheduleManager::UpdateScheduleValues();
+                        int thisHourOfYear = HourOfYear + 1;
+                        if (thisHourOfYear > 8784.0) {
+                            thisHourOfYear = 1;
+                        }
+                        EXPECT_EQ(thisHourOfYear, ScheduleManager::LookUpScheduleValue(1, DataGlobals::HourOfDay, DataGlobals::TimeStep));
+                        EXPECT_EQ(thisHourOfYear, ScheduleManager::Schedule(1).CurrentValue);
+                        EXPECT_EQ(thisHourOfYear, ScheduleManager::GetCurrentScheduleValue(1));
+                    }
+                }
+            }
+        }
+
+        EXPECT_EQ(8784.0, HourOfYear);
+    }
+}
+
+TEST_F(EnergyPlusFixture, Schedule_GetCurrentScheduleValue_DST_RampUp_NoLeap) {
+
+    // So here we'll mimic using a Schedule:Compact that ramps up constantly
+
+    // Schedule:Compact,
+    //   RampingUp,                 !- Name
+    //   Any Number,                !- Schedule Type Limits Name
+    //   Through: 1/1,
+    //   For: AllDays,
+    //    Until: 1:00, 1,
+    //    Until: 2:00, 2,
+    //    [...]
+    //    Until: 24:00, 24,
+    //   Through: 1/2,
+    //   For: AllDays,
+    //    Until: 1:00, 25,
+    //    Until: 2:00, 26,
+    //    Until: 3:00, 27,
+    //    [...]
+    //   Through: 12/31,
+    //   For: AllDays,
+    //    Until: 1:00, 8737,
+    //    [...]
+    //    Until: 24:00, 8760.0;
+
+    // # 'THROUGH" => Number of additional week schedules
+    // # 'FOR' => Number of additional day schedules
+    // So we use 366 Week Schedules, all with one day (LeapYear)
```

Comment:
Test a ramp up schedule for 8760 hourly values (no leap year), checking every hour in the DST and non DST case.

Reply:
This was (perhaps weirdly) harder than the leap case since ScheduleManager always assumes Leap Year in its internal structure. Day 60 (ordinal day corresponding to Feb 29) will just NOT be used if it's not a leap year.